src/main/java/net/ccbluex/liquidbounce/interfaces/GlobalEntityAddition.java

```diff
@@ -0,0 +1,10 @@
+package net.ccbluex.liquidbounce.interfaces;
+
+import net.minecraft.util.math.Vec3d;
+
+public interface GlobalEntityAddition {
+    Vec3d liquidBounce$getActualPosition();
+    boolean liquidBounce$getPassedFirstUpdate();
+    void liquidBounce$updateFirstUpdate();
+    void liquidBounce$setActualPosition(Vec3d newPosition);
+}
```

Comment:
This should be implemented with a class and a hash map. Please try to keep the implementation local to the module file.

The current implementation is hard to understand and undocumented.

Reply:
I think doing it that way is more of a hassle than this. This is a much more cleaner approach as it directly connects to the `MixinEntity` class.

About the documentation, I could write a few things about it to make it easier to understand.

If you still think a class and a hash map could do a better job than this, sure. Why not.

Reply:
Just commented a few things about each function. Should help clear up any confusion, and probably even help someone else reading this learn a new thing.

Reply:
I think I will rather implement Backtrack to use [PacketQueueManager] and provide the necessary utilities from there, similar to what we already do with the actual position of our own player.